### ISSUE TITLE
Confusion Regarding Transferable Targeted Attack Implementation in Appendix L

### ISSUE THREAD

**Issue by molarsu** [WindVChen/DiffAttack#24]
I really appreciate the work you’ve done in your paper. It’s clear that a lot of thought and effort went into the research. 

And I have a question regarding the implementation of the **Transferable Targeted Attack** described in Appendix L of the paper.

In the appendix, it states that DiffAttack, originally designed for untargeted attacks, was adapted for targeted attacks by **removing the transfer loss**  as described in _Section 3.3_. I am curious about how this change alone enables DiffAttack to function as a t**argeted attack** method.

Could you please provide more details or clarify how removing the transfer loss specifically facilitates the implementation of targeted attacks? Are there any additional modifications or considerations that are necessary beyond this adjustment?

Thank you very much for your assistance!

**Comment by WindVChen** [comment]
Hi @molarsu ,

Thank you for recognizing our work, and I apologize for the delayed response—it's been a bit challenging to find time to address GitHub issues these days.

The adaptation process is quite intuitive. The key difference between a targeted attack and an untargeted attack lies in the objective: a targeted attack requires the adversarial example to be classified as a specific category (e.g., dog to cat), whereas an untargeted attack only requires the classification result to be incorrect (e.g., dog could be classified as cat, bird, monkey, etc.).

To convert an untargeted classification attack into a targeted one, the classification loss function needs to be modified. In an untargeted attack, the loss function is designed to maximize the cross-entropy between the true label and the predicted label, ensuring that the classifier produces any prediction except the true one. In contrast, for a targeted attack, the loss function is adjusted to minimize the distance between the targeted label (chosen by the attacker, not the true label) and the predicted label, pushing the classifier to produce the targeted label.

Based on this understanding, we adapted our method for a targeted attack with two primary modifications. First, we adjusted the classification loss as explained above. Second, we removed the transfer loss, but retained the structure loss (as described in Sec. 3.4) to prevent structural distortion. The reason for removing the transfer loss is that it aims to deviate the cross-attention map of the true label, which may be more suitable for untargeted attacks rather than targeted ones. However, you might consider modifying the transfer loss to focus attention on the object associated with the targeted label in a targeted attack—though I haven't tested this, so I can't say for certain if it will work. But if you're interested, it might be worth exploring.

I hope this clarifies your questions.
